README.md

```diff
@@ -442,23 +444,89 @@ We have a small collection of algorithms, one for every day of the month. Scroll
     - Cryptographic hash functions also play a crucial role in blockchains.
-    
+
+
   - **Resources**
     - [Hash function – Wikipedia](https://en.wikipedia.org/wiki/Hash_function)
     - [Hash table - Wikipedia](https://en.wikipedia.org/wiki/Hash_table)
     - [An in-depth video about cryptocurrencies, blockchain and hash functions](https://youtu.be/bBC-nXj3Ng4)
-    
+
+
+### **December 20 - Dijkstra's Shortest Reach**
+  - **Terminologies Involved**
+    - A **graph** is a pictorial representation of a set of objects where some pairs of objects are connected by links.
+    - Each node of the graph is represented as a **vertex**.
+    - An **edge** represents a path between two vertices or a line between two vertices.
+    - A **path** indicates the direction in which information is moving.
+    - **Distance** is the amount of time taken for information transmission between two nodes.
+  - **Problem**
+    - Given an undirected graph and a starting node, determine the lengths of the shortest paths from the starting node to all other nodes in the graph. If a node is unreachable, its distance is `-1` or `INFINITE`. Nodes will be numbered consecutively from to, and edges will have varying distances or lengths.
+    For example, consider the following graph of 9 nodes:
+
+     <img src="/src/docs/Dec20-Example.jpg" width="550" height="250"/>
+
+    - Write an algorithm that returns an array of integers that represent the shortest distance to each node from the start node in ascending order of node number, given the number of nodes, the edges and starting node number.
+
+  - **Example**
+    ```
+    {0, 4, 0, 0, 0, 0, 0, 8, 0}
+    {4, 0, 8, 0, 0, 0, 0, 11, 0}
+    {0, 8, 0, 7, 0, 4, 0, 0, 2}
+    {0, 0, 7, 0, 9, 14, 0, 0, 0}
+    {0, 0, 0, 9, 0, 10, 0, 0, 0}
+    {0, 0, 4, 14, 10, 0, 2, 0, 0}
+    {0, 0, 0, 0, 0, 2, 0, 1, 6}
+    {8, 11, 0, 0, 0, 0, 1, 0, 7}
+    {0, 0, 2, 0, 0, 0, 6, 7, 0}
+    ```
+    ```
+    Vertex   Distance from Source
+    0                0
+    1                4
+    2                12
+    3                19
+    4                21
+    5                11
+    6                9
+    7                8
+    8                14
+    ```
+  - **Resources**  
+    - [What is Dijkstra's Algorithm?](https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm)
+    - [Visualisation of Dijkstra's Algorithm #1](https://visualgo.net/en/sssp)
+    - [Visualisation of Dijkstra's Algorithm #2](https://www.cs.usfca.edu/~galles/visualization/Dijkstra.html)
+
+
+## **December 21 - Currency Convertor**
+  - **Problem**
+      -  Calculate the currency exchange rates between two user input countries using the attached **.csv** file in the reference.
+      - ``Note:`` All the given ``VALUES`` in the csv file are a country's equivalent value in USD.
+      - ``Optional:`` Also, generate a complete list of currency exchange rates between the input currency and the world currencies.
```

Comment:
Add this to the optional part. "For instance, if I input India, then my output would be a list of each country's exchange rate for India. So USA = 71.93, Malaysia = 17 etc.